statsd/statsd.go

```diff
@@ -678,7 +691,13 @@ func (c *Client) Close() error {
 	c.wg.Wait()
 
 	c.Flush()
-	return c.sender.close()
+
+	if err := c.sender.close(); err != nil {
+		return err
+	}
+
+	c.isClosed = true
+	return nil
```

Comment:
Since we closed the workers and aggregators we can consider that the client is closed anyway event if the sender returns an error (as it's not longer capable or receiving points).

```suggestion
        c.isClosed = true
        return c.sender.close()
```